packages/store/ts/codegen/field.ts

```diff
@@ -82,27 +82,47 @@ export function renderFieldMethods(options: RenderTableOptions) {
       const dynamicSchemaIndex = schemaIndex - options.staticFields.length;
 
       if (options.withGetters) {
-        result += renderWithFieldSuffix(options.withSuffixlessFieldMethods, field.name, (_methodNameSuffix) =>
-          renderWithStore(
-            storeArgument,
-            ({ _typedStore, _store, _commentSuffix, _methodNamePrefix }) => `
+        if (field.typeWrappingData && field.typeWrappingData.kind === "staticArray") {
```

Comment:
I don't love this check as is, we repeat it for `pop` and `push`, then again for `length`.

It's also weird to [treat static arrays as dynamic](https://github.com/latticexyz/mud/blob/f6133591a86eb169a7b1b2b8d342733a887af610/packages/store/ts/codegen/userType.ts#L169) with the `if (field.isDynamic)` check, but then differentiate them using this.

Reply:
curious if @dk1a has any insight into this

Reply:
terminology is weird here
`isDynamic`, as well as dynamic and static table fields is a mud-specific thing
A better name for `isDynamic` might be `isNonPrimitive` - it indicates any array or struct
The word `static` in static arrays is used in an unrelated context, taken from solidity

Reply:
https://github.com/latticexyz/mud/issues/1668 is related, I also don't like how hacky this check is, especially since now it's used in more places